bindings/src/Capgemini.PowerApps.SpecFlowBindings/Steps/EntitySubGridSteps.cs

```diff
@@ -59,10 +60,10 @@ public static void WhenISelectAllInTheSubgrid(string subGridName)
         /// </summary>
         /// <param name="index">The position of the record.</param>
         /// <param name="subGridName">The name of the subgrid.</param>
-        [When(@"I open the record at position '(\d+)' in the '(.*)' subgrid")]
-        public static void WhenIOpenTheRecordAtPositionInTheSubgrid(int index, string subGridName)
+        [When(@"I open the record at position (\d[a-z]+) in the '(.*)' subgrid")]
```

Comment:
A few comments on this -

- Changing the existing regular expression here would be a breaking change for any projects currently using this binding. 
- The addition of the ordinal suffixes makes the sentence grammatically incorrect. It would probably be better to add an additional `When` attribute with it reworded to capture something more like `When I open the 1st record in the 'x' subgrid`
- It's an unlikely scenario but this regex breaks for any positions greater than single figures (only 0-9 is possible)
- Might be better to explicitly capture the different ordinal suffixes e.g. `\d+((st)|(nd)|(rd)|(th))`

Reply:
Thanks @ewingjm Updated as per the suggested improvements.  Added new tests for these new bindings or will delete them if not needed

Reply:
All looking good now @thathi. Might be better to delete those tests as the underlying code is the same so it won't affect our code coverage not having them but will lengthen the build pipeline considerably 🙂

Reply:
That make sense. deleted.